src/js/utils/specs/html-utils-spec.js

```diff
@@ -0,0 +1,106 @@
+                        '<a href="world">Rock &amp; Roll</a>'
+                    ]
+                }, function(input, expectedString) {
+                    var result = HtmlUtils.escape(input);
+                    expect(result.toString()).toBe(expectedString);
+                });
+            });
+
+            describe('interpolateHtml', function() {
+                it('can interpolate a string with no parameters provided', function() {
+                    expect(HtmlUtils.interpolateHtml('Hello, world').toString()).toBe(
+                        'Hello, world'
+                    );
+                });
+
+                SpecHelpers.withData({
+                    'can interpolate a string with empty parameters': [
+                        'Hello, world', {},
+                        'Hello, world'
+                    ],
+                    'can interpolate a string with one parameter': [
+                        'Hello, {name}', {name: 'Andy'},
+                        'Hello, Andy'
+                    ],
+                    'does not interpolate additional curly braces': [
+                        'Hello, {name}. Here is a { followed by a }', {name: 'Andy'},
+                        'Hello, Andy. Here is a { followed by a }'
+                    ],
+                    'escapes characters in the template': [
+                        'Rock & Roll', {},
+                        'Rock &amp; Roll'
+                    ],
+                    'does not escape HTML parameters': [
+                        'Hello, {anchor}', {anchor: HtmlUtils.HTML('<a href="world">world</a>')},
+                        'Hello, <a href="world">world</a>'
+                    ],
+                    'escapes characters in parameters': [
+                        'I love {name}', {name: 'Rock & Roll'},
+                        'I love Rock &amp; Roll'
+                    ],
+                    'does not double escape when chaining interpolate calls': [
+                        'I love {name}', {name: HtmlUtils.interpolateHtml('Rock & Roll')},
+                        'I love Rock &amp; Roll'
+                    ],
+                    'full example': [
+                        'You are enrolling in {spanStart}{courseName}{spanEnd}',
+                        {
+                            courseName: 'Rock & Roll',
+                            spanStart: HtmlUtils.HTML('<span class="course-title">'),
+                            spanEnd: HtmlUtils.HTML('</span>')
+                        },
+                        'You are enrolling in <span class="course-title">Rock &amp; Roll</span>'
+                    ]
+                }, function(template, options, expectedString) {
+                    var result = HtmlUtils.interpolateHtml(template, options);
+                    expect(result.toString()).toBe(expectedString);
+                });
+            });
+
+            describe('template', function() {
```

Comment:
Can you add a test for the fact that it is returning an HtmlSnippet that should not get escaped again if wrapped with another call like HtmlUtils.escape()?